### PR TITLE
Fix rust clippies and evm balance length

### PR DESCRIPTION
**Pull request by seanyoung**

### CODE REVIEW

**Comment by seanyoung** [comment]
> This PR removes code so the coverage % drops.

I've deleted unused code and now codecov is happy

**Review by xermicus — APPROVED** [review]

File: src/codegen/cfg.rs [hunk 1]
```diff
@@ -1244,12 +1243,12 @@ impl ControlFlowGraph {
                 gas,
                 salt,
                 value,
-                address,seeds,
+                address,
                 accounts,
                 constructor_no,
                 loc:_
             } => format!(
-                "%{}, {} = constructor(no: {}) salt:{} value:{} gas:{} address:{} seeds:{} {} encoded buffer: {} accounts: {}",
```
[the remark this answers is not in the document]
Comment (seanyoung) [reply src/codegen/cfg.rs:1252]: seeds cannot be used with constructors

[the remark this answers is not in the document]
Comment (seanyoung) [reply src/codegen/cfg.rs:1252]: You removed this functionality

[the remark this answers is not in the document]
Comment (LucasSte) [reply src/codegen/cfg.rs:1252]: > seeds cannot be used with constructors

I can still do `Contract.new{seeds: my_seeds}()`, can't I? What if we are signing with a PDA?

[the remark this answers is not in the document]
Comment (seanyoung) [reply src/codegen/cfg.rs:1252]: In Solidity that's an external call, not a constructor. A constructor looks like `new A(a, b, c);`

[the remark this answers is not in the document]
Comment (LucasSte) [reply src/codegen/cfg.rs:1252]: `Instr::Contructor` is still used if the contract does not have a declared constructor.

https://github.com/hyperledger/solang/blob/90b3f7085fe6375e47a29fe01e802f6118c3ad0e/src/sema/expression/function_call.rs#L3010-L3017

`Expression::Constructor` creates an `Instr::Contructor`, which may contain seeds.

[the remark this answers is not in the document]
Comment (seanyoung) [reply src/codegen/cfg.rs]: ah I did not realise that.

This code is totally broken if seeds are provided.

This PR no longer removed the broken code.

**Review by LucasSte — APPROVED** [review]

**Pull request merged**